flang/include/flang/Optimizer/Transforms/AffinePromotion.h

```diff
@@ -0,0 +1,39 @@
+//===-- Optimizer/Transforms/AffinePromotion.h ------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef OPTIMIZER_TRANSFORMS_AFFINE_PROMOTION_H
+#define OPTIMIZER_TRANSFORMS_AFFINE_PROMOTION_H
+
```

Comment:
Looks like this is making references to types that aren't explicitly included or declared (e.g. llvm::Optional).

Reply:
What about fir and mlir classes? Should I add includes to the header instead of cpp for those as well?

Reply:
Yes. We've been "spelling it all out" as a general practice. llvm::ArrayRef, mlir::Value, fir::LoopOp, etc.

As far as includes, LLVM suggests "as little as possible". https://llvm.org/docs/CodingStandards.html#include-as-little-as-possible
So you'll see some forward references in our .h files (if just a name is needed) and sometimes a `#include` to pull in class decls.